src/client_side.cc

```diff
@@ -2672,6 +2690,9 @@ clientNegotiateSSL(int fd, void *data)
     } else {
         debugs(83, 5, "FD " << fd << " has no certificate.");
     }
+#else
+    debugs(1, DBG_CRITICAL, "ERROR: clientNegotiateSSL session logics not implemented.");
```

Comment:
I have split this #ifdef into two sections and corrected the debugs text. Also dropped it to level 2. Its primary use is for dev testing the handshake process to see what has or has not taken place in a test transaction.

The session reuse AFAIK is purely a performance optimisation for TLS to avoid CPU delays. We can go without that initially. It is high on the list of things to get going though. The ifdef split has exposed the generic call to retrieve session info for logging - more to implement there of course too.

Client certificates are not a commonly used feature and the config parser should be preventing its configuration way before this code comes into play.